### PR TITLE
stderred: init

### LINKED ISSUE #123057: stderred
**Project description**
Makes stderr red through LD_PRELOAD.

**Metadata**

* homepage URL: https://github.com/sickill/stderred
* source URL: https://github.com/sickill/stderred
* license: MIT
* platforms: unix, linux, darwin

### PR DESCRIPTION
**Pull request by spencerpogo**
###### Motivation for this change

Adds the stderred package to nix.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
  - [ ] NixOS
  - [ ] macOS
  - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
  - ```LD_PRELOAD=`readlink result`/lib/libstderred.so python -c 'import sys;print("test",file=sys.stderr)'``` prints red text
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Scoder12** [comment]
Closes #123057

**Comment by legendofmiracles** [comment]
All package related commits should be squashed together, and the commit names don't fit to [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md)

**Comment by Scoder12** [comment]
Commit message should be good now

**Comment by Scoder12** [comment]
...and I messed it up again.

**Comment by davidak** [comment]
Try this to squash the commits:

`git rebase -i HEAD~4`

File: pkgs/tools/misc/stderred/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      functions (fwrite, fprintf, error...) from libc in order to colorize
+      all stderr output that goes to terminal thus making it distinguishable
+      from stdout. Basically it wraps text that goes to file with descriptor
+      "2" with proper ANSI escape codes making text red.
+      It's implemented as a shared library and doesn't require recompilation
+      of existing binaries thanks to preload/insert feature of dynamic
+      linkers.
+    '';
+
+    license = licenses.mit;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/stderred/default.nix:33]:
```suggestion
    description = "stderr in red";
    longDescription = '' stderred hooks on write() and a family of stream
      functions (fwrite, fprintf, error...) from libc in order to colorize
      all stderr output that goes to terminal thus making it distinguishable
      from stdout. Basically it wraps text that goes to file with descriptor
      "2" with proper ANSI escape codes making text red.
      It's implemented as a shared library and doesn't require recompilation
      of existing binaries thanks to preload/insert feature of dynamic
      linkers.
    '';
    license = licenses.mit;
```

**Comment by SuperSandro2000** [comment]
Please follow the contributing guide when creating your PRs. The PR should contain 2 commits.

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -8963,6 +8963,16 @@
     githubId = 59476;
     name = "Peter Schuller";
   };
+  scoder12 = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:8966]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

**Comment by Scoder12** [comment]
oh god what have I done

**Comment by Scoder12** [comment]
I hate this so much. I don't think I have the time or mental bandwidth to fix this PR.

**Review by legendofmiracles — COMMENTED** [review]
LGTM
